File: admin/frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  total_users: number;
  active_users: number;
  users_by_country: Array<{ country: string; count: number }>;
  popular_models: Array<{ id: string; count: number }>;
  popular_hardware: Array<{ id: string; count: number }>;
  popular_usecases: Array<{ id: string; count: number }>;
  device_breakdown: Array<{ device: string; count: number }>;
  browser_breakdown: Array<{ browser: string; count: number }>;
  api_usage_timeline: Array<{ date: string; count: number }>;
  custom_slos_usage: number;
}

export interface SystemHealth {
  metrics: {
    [key: string]: {
      value: number;
      unit: string;
      timestamp: string;
    };
  };
  avg_response_time: number;
  error_count: number;
  status: string;
}

export interface AuditLog {
  id: number;
  admin_id: number;
  action: string;
  resource_type: string;
  resource_id: string;
  old_value: any;
  new_value: any;
  ip_address: string | null;
  timestamp: string;
}

export interface Feedback {
  id: number;
  session_id: string | null;
  feedback_type: string;
  category: string;
  rating: number | null;
  title: string;
  message: string;
  email: string | null;
  screenshot_url: string | null;
  status: string;
  priority: string;
  created_at: string;
  updated_at: string;
  responses: FeedbackResponse[];
}

export interface FeedbackResponse {
  id: number;
  feedback_id: number;
  admin_id: number;
  message: string;
  is_internal: boolean;
  created_at: string;
}

File: admin/backend/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Dict, Optional, Any
from datetime import datetime

class AnalyticsSummary(BaseModel):
    total_users: int
    active_users: int
    users_by_country: List[Dict[str, Any]]
    popular_models: List[Dict[str, Any]]
    popular_hardware: List[Dict[str, Any]]
    popular_usecases: List[Dict[str, Any]]
    device_breakdown: List[Dict[str, Any]]
    browser_breakdown: List[Dict[str, Any]]
    api_usage_timeline: List[Dict[str, Any]]
    custom_slos_usage: int

class SystemHealth(BaseModel):
    metrics: Dict[str, Any]
    avg_response_time: float
    error_count: int
    status: str

class AuditLogEntry(BaseModel):
    id: int
    admin_id: int
    action: str
    resource_type: str
    resource_id: str
    old_value: Optional[Any] = None
    new_value: Optional[Any] = None
    ip_address: Optional[str] = None
    timestamp: datetime
    
    class Config:
        from_attributes = True

class AuditLogResponse(BaseModel):
    total: int
    logs: List[AuditLogEntry]
    limit: int
    offset: int

class FeedbackCreate(BaseModel):
    session_id: Optional[str] = None
    feedback_type: str
    category: str
    rating: Optional[int] = None
    title: str
    message: str
    email: Optional[str] = None
    screenshot_url: Optional[str] = None

class FeedbackUpdate(BaseModel):
    status: Optional[str] = None
    priority: Optional[str] = None

class FeedbackResponseCreate(BaseModel):
    message: str
    is_internal: bool = False

class Feedback(BaseModel):
    id: int
    session_id: Optional[str]
    feedback_type: str
    category: str
    rating: Optional[int]
    title: str
    message: str
    email: Optional[str]
    screenshot_url: Optional[str]
    status: str
    priority: str
    created_at: datetime
    updated_at: datetime
    responses: List[Any] = []
    
    class Config:
        from_attributes = True